Базовая стилизация всех страниц

Artem-Sobolev91 opened the pull request:
Добрый день . Есть проблема с цветом текста и размером в заголовках и параграфах отдельных блоков . Мне кажется есть проблемы наследование или каскадности. Сейчас вроде все сделал, но не уверен, что это правильные решения. Много раз переделывал в advantages (3 цвета букв у блока белый, у заголовков черный, у параграфа светло--черный )  ,много времени ушло. Есть ли какие то ошибки и какой метод выбрать для задания стилей , чтобы просто избегать этого.  Где то у меня логика сломалась.  Можно голосовую консультацию , хочу понять логику( может какие хитрости есть).

---
:mortar_board: [Базовая стилизация всех страниц](https://up.htmlacademy.ru/htmlcss-individual/1/user/2518753/tasks/6)

:boom: https://htmlacademy-htmlcss.github.io/2518753-sedona-1/4/

Artem-Sobolev91 committed:
подключает шрифты,стили в боди

Artem-Sobolev91 committed:
добовляет стили к шапке

Artem-Sobolev91 committed:
добавдяет фон блока hero

Artem-Sobolev91 committed:
Применяет стили к  hero

Artem-Sobolev91 committed:
Добовляет стили в advantages

Создал див, как есть правильно или обернуть вместе с конт. изображением?

Artem-Sobolev91 committed:
Стилизует services-item

Artem-Sobolev91 committed:
Стилизует  search-hotel

Artem-Sobolev91 committed:
Update styles.css

Artem-Sobolev91 committed:
Стилизует footer

Artem-Sobolev91 committed:
Update styles.css

Artem-Sobolev91 committed:
Стилезует фильтры

Artem-Sobolev91 committed:
Стилизует карточки гостиниц

Artem-Sobolev91 committed:
Стилизует pagination

Artem-Sobolev91 committed:
Стилизует subscription-catalog

Artem-Sobolev91 committed:
Update styles.css

Artem-Sobolev91 committed:
Выравнивает строки

Artem-Sobolev91 committed:
Коректирует цвета

Проблема с цветами загаловки color: #000000; и    параграфы color: #333333;
вроде исправил, но как то мне кажется "не правильно-через костыли"

Artem-Sobolev91 committed:
Update index.html

Artem-Sobolev91 committed:
реализую через двойной класс, убираю div

цвета так и прыгают. то заголовок то параграф белый становится. Где то с наследованием напутал.

Artem-Sobolev91 committed:
Убирает лишние div

Artem-Sobolev91 committed:
Update styles.css

Artem-Sobolev91 committed:
Update styles.css

Artem-Sobolev91 committed:
Убирает точки в списках

Artem-Sobolev91 committed:
Update styles.css

Artem-Sobolev91 committed:
Update index.html

andreysgra commented on catalog.html:
```diff
@@ -42,162 +43,168 @@
-            <a class="pagination-link" href="#">10</a>
-          </li>
-        </ol>
+      <div class="filter-background">
+        <header class="inner-header">
+          <h2 class="inner-header-title">Гостиницы Седоны</h2>
+          <ul class="breadcrumbs-item">
+            <li class="breadcrumbs-item">
+              <a class="breadcrumbs-link" href="index.html">
+                <span class="visually-hidden">Главная</span>
+              </a>
+            </li>
+            <li class="breadcrumbs-item">
+              <a class="breadcrumbs-link">Гостиницы</a>
+            </li>
+          </ul>
+        </header>
+        <section class="catalog-products">
```
Разметку блока "Гостиницы Седоны" стоит переделать. Вы добавили `div` с классом `.filter-background`, но кроме фонового изображения он ничего полезного не делает, а изображение можно перенести на другой элемент. В результате этот `div` является лишним элементом разметки (критерий [HTML-06](https://up.htmlacademy.ru/htmlcss-individual/1/criteries#HTML-06)).

У вас есть `section` с классом `.catalog-products`, но в нём только форма, хотя в эту секцию можно пометить и блок `.inner-header`. Это вполне логично, так как данный блок размечен тегом `header`. В нём же есть заголовок в явном виде "Гостиницы Седоны" (кстати для него лучше использовать `h1`, он вполне подходит под заголовок страницы сайта), а скрытый заголовок "Список гостиниц с фильтрами" просто удаляете за ненадобностью.

Итак, что стоит сделать (описанное здесь немного отличается от написанного выше):
1. `div` с классом `.filter-background` удаляете.
2. Заголовок "Гостиницы Седоны" переносите в секцию `.catalog-products`, тег `h2` меняете на `h1`. Заголовок будет идти сразу после тега `section`, из `header` его убираете.
3. В `header` с классом `.inner-header` остаётся только список `.breadcrumbs-item` и приходим к выводу, что `header` не нужен и его можно удалить (но если так хочется, можете оставить).
4. Фоновое изображение `images/subscribe-2.jpg` применяем к секции `.catalog-products`.

Artem-Sobolev91 replied:
Заменил div на <section class="filter-background"> , потому что у меня фон подкладывался и под карточки товаров.  Не правильно походу секции разметил, и от этого див влепил не по делу.

andreysgra commented on catalog.html:
```diff
@@ -42,162 +43,168 @@
+            <button class="button" type="reset">Сбросить</button>
+          </form>
+      </div>
+      <p class="page-search">Найдено гостиниц: 38</p>
```
Количество найденных гостиниц лучше обернуть в `span`, так как это значение в реальном сайте изменяется динамически, то удобнее его подставлять в отдельном теге:
`<p class="page-search">Найдено гостиниц: <span>38</span></p>`

andreysgra commented on catalog.html:
```diff
@@ -42,162 +43,168 @@
+          </div>
+        </li>
+      </ul>
+      <ol class="pagination">
```
Не вижу необходимости здесь использовать `ol`, так как нумерация прерывается и после ссылки `...` (а это уже не нумерация) идет число 10. Так что этот список никак нельзя назвать нумерованным.

andreysgra commented on index.html:
```diff
@@ -1,137 +1,138 @@
+    </nav>
+  </header>
+  <main class="main-index">
+    <section class="hero">
+      <h1><span class="visually-hidden">Sedona</span></h1>
+      <img src="images/welcome-sedona.svg" width="458" height="352" alt="добро пожаловать в Седону">
+    </section>
+    <section class="hero-advantages">
+      <h2 class="visually-hidden">Sedona</h2>
+      <p class="lead">Седона — небольшой городок в Аризоне,заслуживающий большего!</p>
+      <p>Рассмотрим причины, по которым Седона круче, чем Гранд-Каньон!</p>
+    </section>
```
Если посмотреть заголовки на странице, то увидим такую картину:
![image](https://github.com/htmlacademy-htmlcss/2518753-sedona-1/assets/17790771/28165831-a24d-44cb-a1bc-ba0c6fd6e202)
Подряд идут два одинаковых заголовка. Здесь вполне достаточно одного заголовка - в секции `hero`, но стоит текст изменить на более информативный. От второго заголовка можно избавиться, если тег `section` заменить на `div`. Собственно тегом `section` размечается более-менее значимый раздел страницы, а этот на такое значение не претендует.

Artem-Sobolev91 replied:
Понял. да так правильнее. Див мы используем где нет смысла , просто обертку, а секцию где смысловая нагрузка(контент, текст , товары и тд). Вот эту разницу не уловил тогда, поэтому и дал название сайту, а потом название секции. Еще с этапа разметки тянуться проблемы. Там я значит  тоже имею пробелы в знаниях. Наверное как по макету понять какие секции и какой тек поставить( секция, параграф и тд.)

andreysgra commented on index.html:
```diff
@@ -1,137 +1,138 @@
+    <section class="advantages">
+      <ul class="advantages-list">
+        <li class="advantages-item advantages-item-city">
+          <h3>Настоящий городок</h3>
```
Чтобы не нарушилась иерархия заголовков (в предыдущем комментарии я предлагал `h2` Sedona удалить), в секции `Sedona` нужно добавить скрытый заголовок `h2`.

Artem-Sobolev91 replied:
Понял, то есть у каждой секции(всегда) должен быть заголовок.

andreysgra commented on index.html:
```diff
@@ -1,137 +1,138 @@
-    </footer>
-  </body>
+    </section>
+    <section class="search-hotel">
```
На мой взгляд `section` стоит заменить на `div`, а текст "Заинтересовались?" мало похож на заголовок, лучше использовать `p`.

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,415 @@
+  line-height: 21px;
+  background: #fff;
+  color: #333333;
+}
```
Попробую ответить на ваши вопросы.

Значения некоторых свойств наследуются от родительского элемента к дочернему. К таким свойствам относится `font-family`, `font-size`, `line-height`, `color` и ряд других. Поскольку тег `body` является тегом почти самого верхнего уровня, то заданные в нём значения перечисленных свойств будут унаследованы всеми тегами, которые лежат внутри него вне зависимости от уровня вложенности.

В данном проекте используется только один тип шрифта - PT Sans, поэтому указав его в `body` нет необходимости указывать его для других элементов. Например в `.navigation-link` вообще можно не указывать это свойство. Исключения есть только для тегов `input`, `button`, `textarea`, `select`. Для них значение свойства `font-family` нужно указывать явно или воспользоваться ключевым словом `inherit`: `font-family: `inherit`.

Цвет текста, свойство `color`, не наследуется у ссылок и если у ссылки цвет текста такой же, как и у родительского элемента, то также можно использовать слово `inherite`: `color: inherit`. Если цвет отличается, то указываем его явно.

Какие значения свойств `font-size`, `line-height`, `color` стоит задать в `body`: указываются значения для основного текста на сайте. Таким текстом набраны статьи. В данном макете статей нет, но за основу можно взять текст из блоков преимуществ.

Все остальные значения свойств для текста нужно будет переопределить. Заголовки в макете имеют чёрный цвет, а также у них отличается `font-size` и `line-height`.

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,415 @@
+
+
+.page-header {
+  background: #fff;
```
[Формат цветов](https://codeguide.academy/html-css.html#css-color-format): _Шестнадцатеричное значение цвета не сокращено, а записано полностью из всех шести символов._

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,415 @@
+}
+
+.navigation-link {
+  font-family: "PT Sans", sans-serif;
```
Здесь и в других классах: как я написал выше, свойство `font-family` можно не указывать.

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,415 @@
+  font-size: 20px;
+  line-height: 24px;
+  color: #000000;
+  background-color: #fff;
```
[Формат цветов](https://codeguide.academy/html-css.html#css-color-format): _Шестнадцатеричное значение цвета не сокращено, а записано полностью из всех шести символов._

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,415 @@
+  font-family: "PT Sans", sans-serif;
+  font-size: 16px;
+  line-height: 24px;
+  color: #FFFFFF;
```
[Формат цветов](https://codeguide.academy/html-css.html#css-color-format): _Для записи должны быть использованы строчные буквы._

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,415 @@
+
+.hero {
+  color: #000000;
+  background-color: #fff;
```
[Формат цветов](https://codeguide.academy/html-css.html#css-color-format): _Шестнадцатеричное значение цвета не сокращено, а записано полностью из всех шести символов._

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,415 @@
+.hero {
+  color: #000000;
+  background-color: #fff;
+  background-image: url(../images/index-background.jpg);
```
[Кавычки](https://codeguide.academy/html-css.html#css-quotes): _В необязательных случаях кавычки не опущены. Во всех случаях в стилях использованы двойные кавычки._

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,415 @@
+  background-image: url(../images/index-background.jpg);
+  background-repeat: no-repeat;
+  background-size: 100% auto;
+  width: 1200px;
+  height: 485px;
```
В этом задании давайте пока не касаться геометрии блоков, так как это раздел по построению сетки сайта. Сделав сейчас что-то неправильно, в дальнейшем придётся исправлять. Поэтому эти свойства удалите.

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,415 @@
+.hero-advantages {
+  font-family: "PT Sans", sans-serif;
+  color: #000000;
+  background-color: #fff;
```
[Формат цветов](https://codeguide.academy/html-css.html#css-color-format): _Шестнадцатеричное значение цвета не сокращено, а записано полностью из всех шести символов._

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,415 @@
+.hero-advantages p {
+  font-family: "PT Sans", sans-serif;
+  color: #333333;
+  background-color: #fff;
```
[Формат цветов](https://codeguide.academy/html-css.html#css-color-format): _Шестнадцатеричное значение цвета не сокращено, а записано полностью из всех шести символов._

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,415 @@
+}
+
+.advantages-item {
+  background-color: #82B3D3;
```
[Формат цветов](https://codeguide.academy/html-css.html#css-color-format): _Для записи должны быть использованы строчные буквы._

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,415 @@
+
+.advantages-item-city,
+.advantages-item-bridge {
+  background-color: #82B3D3;
+  color: #FFFFFF;
```
[Формат цветов](https://codeguide.academy/html-css.html#css-color-format): _Для записи должны быть использованы строчные буквы._

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,415 @@
+}
+
+.advantages-item-city h3 {
+  color: #FFFFFF;
```
[Формат цветов](https://codeguide.academy/html-css.html#css-color-format): _Для записи должны быть использованы строчные буквы._

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,415 @@
+}
+
+.advantages-item-bridge h3 {
+  color: #FFFFFF;
```
[Формат цветов](https://codeguide.academy/html-css.html#css-color-format): _Для записи должны быть использованы строчные буквы._

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,415 @@
+
+.search-hotel a {
+  text-decoration: none;
+  color: #FFFFFF;
```
[Формат цветов](https://codeguide.academy/html-css.html#css-color-format): _Для записи должны быть использованы строчные буквы._

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,415 @@
+}
+
+.subscription {
+  background-image: url(../images/subscribe-1.jpg);
```
[Кавычки](https://codeguide.academy/html-css.html#css-quotes): _В необязательных случаях кавычки не опущены. Во всех случаях в стилях использованы двойные кавычки._

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,415 @@
+  background-repeat: no-repeat;
+  background-size: 100% auto;
+  text-align: center;
+  color: #FFFFFF;
```
[Формат цветов](https://codeguide.academy/html-css.html#css-color-format): _Для записи должны быть использованы строчные буквы._

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,415 @@
+.main-footer {
+  font-family: "PT Sans", sans-serif;
+  color: #000000;
+  background-color: #fff;
```
[Формат цветов](https://codeguide.academy/html-css.html#css-color-format): _Шестнадцатеричное значение цвета не сокращено, а записано полностью из всех шести символов._

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,415 @@
+}
+
+.filter-background {
+  background-image: url(../images/subscribe-2.jpg);
```
[Кавычки](https://codeguide.academy/html-css.html#css-quotes): _В необязательных случаях кавычки не опущены. Во всех случаях в стилях использованы двойные кавычки._

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,415 @@
+  background-image: url(../images/subscribe-2.jpg);
+  background-repeat: no-repeat;
+  background-size: 100% 100%;
+  color: #fff;
```
[Формат цветов](https://codeguide.academy/html-css.html#css-color-format): _Шестнадцатеричное значение цвета не сокращено, а записано полностью из всех шести символов._

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,415 @@
+  font-size: 16px;
+  line-height: 21px;
+  text-decoration: none;
+  color: #FFFFFF;
```
[Формат цветов](https://codeguide.academy/html-css.html#css-color-format): _Для записи должны быть использованы строчные буквы._

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,415 @@
+  font-weight: 400;
+  font-size: 18px;
+  line-height: 21px;
+  border: 1px solid #E6E6E6;
```
[Формат цветов](https://codeguide.academy/html-css.html#css-color-format): _Для записи должны быть использованы строчные буквы._

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,415 @@
+}
+
+.product-card .product-card-button {
+  color: #FFFFFF;
```
[Формат цветов](https://codeguide.academy/html-css.html#css-color-format): _Для записи должны быть использованы строчные буквы._

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,415 @@
+}
+
+.product-card .product-card-button-favorites {
+  color: #FFFFFF;
+  background-color: #82B3D3;
```
[Формат цветов](https://codeguide.academy/html-css.html#css-color-format): _Для записи должны быть использованы строчные буквы._

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,415 @@
+}
+
+.product-card .rating {
+  background-color: #F2F2F2;
```
[Формат цветов](https://codeguide.academy/html-css.html#css-color-format): _Для записи должны быть использованы строчные буквы._

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,415 @@
+  font-size: 20px;
+  line-height: 36px;
+  text-align: center;
+  color: #FFFFFF;
+  background-color: #82B3D3;
```
[Формат цветов](https://codeguide.academy/html-css.html#css-color-format): _Для записи должны быть использованы строчные буквы._

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,415 @@
+
+.subscription-catalog {
+  color: #000000;
+  background-color: #fff;
```
[Формат цветов](https://codeguide.academy/html-css.html#css-color-format): _Шестнадцатеричное значение цвета не сокращено, а записано полностью из всех шести символов._

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,415 @@
+
+.hero-advantages p {
+  font-family: "PT Sans", sans-serif;
+  color: #333333;
```
Так как цвет `color: #333333` указан в `body`, то он наследуется всеми дочерними элементами и указывать его не имеет смысла. Если исследовать этот параграф в Dev Tools, то мы увидим, что значение `body` переопределено точно таким же значением в `.hero-advantages p`, но вы его ещё переопределяли в `.hero-advantages` на чёрный:
![image](https://github.com/htmlacademy-htmlcss/2518753-sedona-1/assets/17790771/252c83cc-77c7-497f-bf44-2b84fa8286f4)

Переопределять цвет текста непосредственно в `.hero-advantages` не стоило. В этом блоке два элемента: заголовок и параграф, но у параграфа цвет текста наследуется из `body` и цвет не совпадает только в заголовке, поэтому логичнее переопределить цвет только заголовка в селекторе `.hero-advantages h2` и не делать лишнюю работу.

Artem-Sobolev91 replied:
Точно! Вот где логика сломалась. Произошло потому что я с начало не правильно цвет Body указал, а когда изменил все "поехало". А началось вообще с того что стили для body не там смотрел. Выше как раз вы про это сказали. По цепочки значит все у меня полетело.

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,415 @@
+  color: #FFFFFF;
+}
+
+.advantages-item-city h3 {
+  color: #FFFFFF;
```
Цвет текста вы задали в классе `.advantages-item-city` (про названия классов поговорим ниже) и это правильно, так как весь текст в этом блоке белый:
![image](https://github.com/htmlacademy-htmlcss/2518753-sedona-1/assets/17790771/07966ad4-086c-4685-b550-6fd5e9dc819b)

Но цвет заголовков всё равно остаётся чёрным, так как в селекторе `.advantages-list h3` задан чёрный цвет, а специфичность селектора `.advantages-list h3` выше, чем `.advantages-item-city` (если ещё не читали про специфичность, то прочитайте обязательно). По этой причине приходится снова переопределять цвет заголовка в селекторе `.advantages-item-city h3`. Можно было бы задать классы для `h3` в синих блоках, но не намного бы мы от этого выиграли.

Теперь об именовании классов. Вижу, что помимо основного класса `.advantages-item` у вас есть дополнительный класс в каждом теге `li`. Дополнительный класс нужен, но не уникальный для каждого блока. Первый и пятый `li` ничем друг от друга не отличаются, следовательно зачем им нужны уникальные классы? Лучше добавить одинаковый класс, который описывал бы отличия от `.advantages-item`, например `.advantages-item-wide`.

Тегам `li`, которые идут со 2-го по 4-ый также не нужны уникальные дополнительные классы. В принципе для них вообще не нужны дополнительные классы, так как цвет фона можно задать в псевдо классе `:nth-child()` (в качестве значения можно использовать ключевые слова `odd` и `even`). Цвет фона в длинных блоках переопределяется в классе `.advantages-item-wide`.

Artem-Sobolev91 replied:
Сложно, но вроде сделал.

Artem-Sobolev91 committed:
Меняет див на секцию

Перенес стили из li  в ul , проверил ссылки нновые добавляются в заданных стилях

Artem-Sobolev91 committed:
Update styles.css

Artem-Sobolev91 committed:
Update catalog.html

Artem-Sobolev91 committed:
Исправляет pagination

Artem-Sobolev91 committed:
Удоляет h2

Artem-Sobolev91 committed:
Добовляет h2

Artem-Sobolev91 committed:
Меняет секцию на див

Artem-Sobolev91 committed:
Update styles.css

Artem-Sobolev91 committed:
Update styles.css

Artem-Sobolev91 committed:
Update styles.css

Artem-Sobolev91 committed:
Update styles.css

Artem-Sobolev91 committed:
Update styles.css

Artem-Sobolev91 committed:
Update styles.css

Artem-Sobolev91 committed:
Редактирует advantages

andreysgra commented on catalog.html:
```diff
@@ -1,236 +1,241 @@
-      </a>
-    </footer>
-  </body>
+      <h3 class="visually-hidden">Список гостиниц с фильтрами</h3>
```
```suggestion
      <h2 class="visually-hidden">Список гостиниц с фильтрами</h2>
```
Иначе нарушается иерархия заголовков.
![image](https://github.com/htmlacademy-htmlcss/2518753-sedona-1/assets/17790771/848e2c29-32a3-43e7-804e-47a1afef636c)

andreysgra commented on catalog.html:
```diff
@@ -1,236 +1,241 @@
+        <a class="pagination-link" href="#">10</a>
+      </li>
+    </ul>
+    </section>
```
Похоже, что закрывающий тег лишний. Ошибка валидации HTML.

andreysgra commented on index.html:
```diff
@@ -1,137 +1,138 @@
-  </body>
+    </section>
+    <div class="search-hotel">
+      <p class="search-hotel-question">Заинтересовались?</h2>
```
```suggestion
      <p class="search-hotel-question">Заинтересовались?</p>
```

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,391 @@
+  color: #333333;
+}
+
+
+
```
Лишние пустые строки. [Структура объявления](https://codeguide.academy/html-css.html#css-structure): _Между блоками правил есть одна пустая строка._

andreysgra commented on styles/styles.css:
```diff
@@ -0,0 +1,391 @@
+  color: #ffffff;
+}
+
+.advantages-list :nth-child(2),
+.advantages-list :nth-child(4) {
+  background-color: rgba(131, 179, 211, 0.12);
+}
+
+.advantages-list :nth-child(3) {
+  background-color: rgba(131, 179, 211, 0.2);
+}
```
Здесь сделано некорректно. Правильное написание селектора такое: `.advantages-item:nth-child(2)`.

Тем не менее, лучше не явно указывать номер элемента, а использовать формулу. Можно предположить глядя на макет, что цвета чередуются: нечётные блоки имеют цвет `rgba(131, 179, 211, 0.2)`, а чётные цвет `rgba(131, 179, 211, 0.12)`. Для обозначения чётного числа есть ключевое слово `even`, для нечётного `odd`. Следовательно, для нечётного блока мы можем написать такой селектор:
```
.advantages-item:nth-child(odd) {
    background-color: rgba(131, 179, 211, 0.2);
}
```

Для чётного блока сделайте по аналогии.

В блоках с классом `.advantages-item-wide` цвет тоже изменится и чтобы этого не произошло нужно усилить селектор класса `.advantages-item-wide` на `.advantages-item.advantages-item-wide`.

Artem-Sobolev91 committed:
Изменяет цвет фона

У меня почему то чередуются теги  внутри <li>

Artem-Sobolev91 committed:
Разбирается с ошибкой

Понял. Прочитал еще раз Ваш комент, и понял. Страдаю не внимательностью. Теперь правильно применил тег и убрал пробел. Все заработало. Спасибо.

Artem-Sobolev91 committed:
Update styles.css

Artem-Sobolev91 committed:
Update styles.css

Artem-Sobolev91 committed:
Изменяет стиль в services-item

псевдокласс :nth-child

Artem-Sobolev91 committed:
Update styles.css
